x-pack/plugins/observability_solution/apm/public/components/app/service_tab_content/index.tsx

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { ReactNode } from 'react-markdown';
+import { EuiSpacer } from '@elastic/eui';
+import { SignalTypes } from '../../../../common/entities/types';
+import { useApmServiceContext } from '../../../context/apm_service/use_apm_service_context';
+import { isApmSignal, isLogsSignal } from '../../../utils/get_signal_type';
+import { useLocalStorage } from '../../../hooks/use_local_storage';
+import { emptyStateContent } from './constants';
+import { Tab } from './types';
+import { ServiceTabEmptyState } from '../service_tab_empty_state';
+import { SearchBar } from '../../shared/search_bar/search_bar';
+
+export interface ServiceTabContentProps {
+  tabName: Tab;
+  children: ReactNode;
+}
+
+export function ServiceTabContent({ children, tabName }: ServiceTabContentProps) {
+  const { serviceEntitySummary } = useApmServiceContext();
+
+  const hasOnlyLogsSignal =
+    serviceEntitySummary?.signalTypes &&
+    !isApmSignal(serviceEntitySummary.signalTypes as SignalTypes[]) &&
+    isLogsSignal(serviceEntitySummary.signalTypes as SignalTypes[]);
+
+  const [dismissedLogsOnlyEmptyState, setdismissedLogsOnlyEmptyState] = useLocalStorage(
+    `apm.dismissedLogsOnlyEmptyState.${tabName}`,
+    false
+  );
+
+  const displayEmptyStateOnly = tabName !== 'overview';
+  const displaySearchBar = tabName === 'service-map';
```

Comment:
I don't think we should show the search bar for an empty service map page.

Reply:
+1.

Reply:
From what I understand, we are supposed to display the search bar in every tab that has it when data is available. The other tabs handle this differently in their templates, so I had to add it here specifically for the service-map tab.

Could you help me understand why you think we shouldn't display it in this tab, while we do show it in the other affected tabs? Except for Infra, which doesn't have it even when there's data.

Reply:
It feels wrong IMHO. Especially because of the `displayEmptyStateOnly` and `displaySearchBar`. This can easily grow to be a very complex and hard to maintain component as more custom logic is added to this HOC.

Reply:
Got it. I'll reconsider the approach and move the logic into each of the affected components. 
Thanks for the feedback.

Regarding the initial topic of the thread:
> I don't think we should show the search bar for an empty service map page.
>> From what I understand, we are supposed to display the search bar in every tab that has it when data is available. The other tabs handle this differently in their templates, so I had to add it here specifically for the service-map tab.
>> 
>> Could you help me understand why you think we shouldn't display it in this tab, while we do show it in the other affected tabs? Except for Infra, which doesn't have it even when there's data.

What are your thoughts? @cauemarcondes

Reply:
Ok, let's remove the search bar from all the new empty states.

The [Figma designs](https://www.figma.com/design/pFID9w03XPtZ4qwisM59sR/ECO-Observability?node-id=8388-28960&t=mnpj3yg1BTm2uWs0-4) included the search bars in the final layouts. We need to refine our process to ensure these details are correctly incorporated next time.